src/Lykke.Service.BlockchainApi.Sdk/PeriodicalHandlers/DepositHandler.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Common;
+using Lykke.Common.Log;
+using Lykke.Service.BlockchainApi.Sdk.Domain;
+
+namespace Lykke.Service.BlockchainApi.Sdk.PeriodicalHandlers
+{
+    public class DepositHandler<T> : TimerPeriod
+    {
+        readonly DepositWalletRepository _depositWallets;
+        readonly OperationRepository _operations;
+        readonly AssetRepository _assets;
+        readonly StateRepository<T> _state;
+        readonly IBlockchainJob<T> _job;
+
+        public DepositHandler(TimeSpan period, ILogFactory logFactory, DepositWalletRepository depositWallets, OperationRepository operations, AssetRepository assets,
+            StateRepository<T> state, IBlockchainJob<T> job) : base(period, logFactory, nameof(DepositHandler<T>))
+        {
+            _depositWallets = depositWallets;
+            _operations = operations;
+            _assets = assets;
+            _state = state;
+            _job = job;
+        }
+
+        public override async Task Execute()
+        {
+            var currentState = await _state.GetAsync();
+            var state = currentState != null ? currentState.State : default(T);
```

Comment:
```suggestion
            var state = currentState?.State ?? default(T);
```

Reply:
It doesn't work for not-restricted generic parameter - https://stackoverflow.com/questions/32580536/operator-cannot-be-applied-to-operand-of-type-t. I don't want to add restrictions because both reference types and value types make sense here and could be used as state.